import-scripts/oncokb-annotator.sh

```diff
@@ -1,174 +1,298 @@
+    ONCOKB_OUT_DIR="$STAGING_DIR/oncokb"
+
+    # scripts
+    MAF_ANNOTATOR_SCRIPT="$ONCOKB_ANNOTATOR_HOME/MafAnnotator.py"
+    FUSION_ANNOTATOR_SCRIPT="$ONCOKB_ANNOTATOR_HOME/FusionAnnotator.py"
+    CNA_ANNOTATOR_SCRIPT="$ONCOKB_ANNOTATOR_HOME/CnaAnnotator.py"
+    CLINICAL_ANNOTATOR_SCRIPT="$ONCOKB_ANNOTATOR_HOME/ClinicalDataAnnotator.py"
+    PDF_SCRIPT="$ONCOKB_ANNOTATOR_HOME/OncoKBPlots.py"
+    README_SCRIPT="$ONCOKB_ANNOTATOR_HOME/GenerateReadMe.py"
+
+    # data files and parameters
+    SOURCE_SAMPLE_FILE="$MSK_SOLID_HEME_DATA_HOME/data_clinical_sample.txt"
+    STAGING_SAMPLE_FILE="$STAGING_DIR/data_clinical_sample.txt"
+    ONCOKB_SAMPLE_FILE="$ONCOKB_OUT_DIR/data_clinical_sample.oncokb.txt"
+    ONCOKB_SOMATIC_SAMPLE_FILE="$ONCOKB_OUT_DIR/data_clinical_sample_somatic.oncokb.txt"
+
+    SOURCE_MAF_FILE="$MSK_SOLID_HEME_DATA_HOME/data_mutations_extended.txt"
+    STAGING_MAF_FILE="$STAGING_DIR/data_mutations_extended.txt"
+    ONCOKB_MAF_FILE="$ONCOKB_OUT_DIR/data_mutations_extended.oncokb.txt"
+    ONCOKB_PREVIOUS_MAF_FILE="$ONCOKB_OUT_DIR/data_mutations_extended.oncokb.previous.txt"
+    ONCOKB_SOMATIC_MAF_FILE="$ONCOKB_OUT_DIR/data_mutations_extended_somatic.oncokb.txt"
+
+    SOURCE_FUSION_FILE="$MSK_SOLID_HEME_DATA_HOME/data_fusions.txt"
+    STAGING_FUSION_FILE="$STAGING_DIR/data_fusions.txt"
+    ONCOKB_FUSION_FILE="$ONCOKB_OUT_DIR/data_fusions.oncokb.txt"
+
+    SOURCE_CNA_FILE="$MSK_SOLID_HEME_DATA_HOME/data_CNA.txt"
+    STAGING_CNA_FILE="$STAGING_DIR/data_CNA.txt"
+    ONCOKB_CNA_FILE="$ONCOKB_OUT_DIR/data_CNA.oncokb.txt"
+
+    ONCOKB_PDF_NUMBER=100
+    ONCOKB_PDF_FILE="$ONCOKB_OUT_DIR/data_clinical_sample.oncokb.pdf"
+    ONCOKB_SOMATIC_PDF_FILE="$ONCOKB_OUT_DIR/data_clinical_sample_somatic.oncokb.pdf"
+
+    ONCOKB_README_FILE="$ONCOKB_OUT_DIR/README"
+
+    echo $(date)
+    echo "Staring oncokb-annotator run on $SOURCE_DIR"
+    ONCOKB_ANNOTATION_SUCCESS=1
+
+    # check if oncokb_annotator clone is up to date and log if not
+    if ! git_clone_is_up_to_date "$ONCOKB_ANNOTATOR_HOME" ; then
+        echo "WARNING : repository clone is out of date at $ONCOKB_ANNOTATOR_HOME"
+    fi
+
+    # Pause, up to some time limit for dmp imports to complete
+    if [ $ONCOKB_ANNOTATION_SUCCESS -eq 1 ] ; then
+        echo $(date)
+        echo "Checking that dmp import pipeline run in complete..."
+        MAXIMUM_WAIT_SECONDS=$(( 6 * 3600 )) # 6 hours wait time
+        wait_for_completion_of_dmp_import_run $MAXIMUM_WAIT_SECONDS
+        if [ $? -ne 0 ] ; then
+            echo "Failed to wait long enough for the dmp import pipeline run to complete, exiting..."
+            ONCOKB_ANNOTATION_SUCCESS=0
+        fi
+    fi
+
+    # Update DMP repo before running annotations
+    if [ $ONCOKB_ANNOTATION_SUCCESS -eq 1 ] ; then
+        echo $(date)
+        echo "Updating the DMP source repository..."
+        fetch_updates_in_data_sources dmp
+        if [ ${#failed_data_source_fetches[*]} -ne 0 ] ; then
```

Comment:
`failed_data_source_fetches` set in `set-data-source-environment-vars.sh` and will contain a list of all data source fetches which failed after calling fetch_updates_in_data_sources()